src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -735,16 +735,34 @@ private bool IsByRefTarget(int slot)
             return false;
         }
 
-        private void ReportWWarning(SyntaxNode syntax)
+        private void ReportNonSafetyDiagnostic(SyntaxNode syntax)
         {
-            ReportDiagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, syntax);
+            ReportNonSafetyDiagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, syntax);
         }
 
-        private void ReportDiagnostic(ErrorCode errorCode, SyntaxNode syntaxNode, params object[] arguments)
+        private void ReportNonSafetyDiagnostic(ErrorCode errorCode, SyntaxNode syntax)
+        {
+            // All warnings should be in the `#pragma warning ... nullable` set.
+            Debug.Assert(!ErrorFacts.NullableFlowAnalysisSafetyWarnings.Contains(MessageProvider.Instance.GetIdForErrorCode((int)errorCode)));
+            Debug.Assert(ErrorFacts.NullableFlowAnalysisNonSafetyWarnings.Contains(MessageProvider.Instance.GetIdForErrorCode((int)errorCode)));
+#pragma warning disable CS0618
+            ReportDiagnostic(errorCode, syntax);
```

Comment:
Why are we using an obsolete API here?

Reply:
Oh I see, you're using `Obsolete` to enforce calling these two instead.

---
In reply to: [242338210](https://github.com/dotnet/roslyn/pull/31806#discussion_r242338210) [](ancestors = 242338210)